types/index.d.ts

```diff
@@ -1,45 +1,98 @@
 //
 // Feel adventurous? Help us fill in the gaps – don't wait for tomorrow!
 // -------------------------------------------------------------------------
 
-export class Agent {
+import { Tool } from "langchain/tools";
+import type { AGENT_STATUS_enum, TASK_STATUS_enum } from "./enums.d.ts";
+import type {
+  BaseAgent,
+  IBaseAgentParams,
+  ILLMConfig,
+  ITaskStats,
+  TAgentTypes,
+  TStore,
+} from "./types.d.ts";
 
-    [key: string]: any;
+declare module "agenticjs" {
+  export class Agent {
+    agentInstance: BaseAgent;
+    type: string;
 
-    constructor(props?: Record<string, any>) {
-        Object.assign(this, props);
-    }    
+    constructor(type: TAgentTypes, config: IBaseAgentParams);
 
-    // Implementation details will be defined later... 
-    // Yes, this is the 'I’ll clean my desk next week' kind of later.
-}
+    createAgent(type: TAgentTypes, config: IBaseAgentParams): BaseAgent;
+    executeTask(task: Task): Promise<any>;
 
-export class Task {
+    setStore(store: TStore): void;
+    setEnv(env: Record<string, any>): void;
+    setStatus(status: AGENT_STATUS_enum): void;
 
-    [key: string]: any;
+    id(): string;
+    name(): string;
+    role(): string;
+    goal(): string;
+    background(): string;
+    tools(): Tool[];
+    status(): AGENT_STATUS_enum;
+    llmConfig(): ILLMConfig;
+    llmSystemMessage(): string;
+    forceFinalAnswer(): boolean;
+  }
 
-    constructor(props?: Record<string, any>) {
-        Object.assign(this, props);
-    }    
+  export interface ITaskParams {
+    title?: string;
+    description: string;
+    expectedOutput: any;
+    agent: BaseAgent;
+    isDeliverable?: boolean;
+  }
 
-    // Implementation details will be defined later... 
-    // Like that gym membership we all get in January.
-}
+  export class Task {
+    id: string;
+    title: string;
+    description: string;
+    expectedOutput: string;
+    isDeliverable: boolean;
+    agent: BaseAgent;
+    status: TASK_STATUS_enum;
+    result: any; // ? Need more context
+    stats: ITaskStats | null;
+    duration: number | null;
+    dependencies: Task[];
+    interpolatedTaskDescription: string | null;
+    store: TStore;
+
+    constructor(params: ITaskParams);
+
+    setStore(store: TStore): void;
+  }
 
-export class Team {
+  export interface ITeamParams {
+    name: string;
+    agents?: BaseAgent[];
+    tasks?: Task[];
+    logLevel?: string;
+    inputs?: Record<string, string>;
+    env?: Record<string, any> | null;
+  }
 
-    [key: string]: any;
+  export class Team {
+    store: TStore;
 
-    constructor(props?: Record<string, any>) {
-        Object.assign(this, props);
-    }    
+    constructor(params: ITeamParams);
 
-    // Implementation details will be defined later... 
-    // We're scheduling a meeting to plan the meeting about this.
+    start(): Promise<void>;
+    getStore(): TStore;
+    useStore(): TStore;
+    subscribeToChanges(
```

Comment:
The listener is a callback function that gets executed when the specified properties change.

Example:

```js
subscribeToChanges((newValues) => {
    console.log('State updated:', newValues);
}, ['user', 'isLoggedIn']);

```

Reply:
This example has helped clarify and solve the type here.